### PR TITLE
Update [CPP]cookie_collect_csubhradipta.cpp

### PR DESCRIPTION
**Pull request by cSubhradipta**
Added code for cookie_collect in cpp

- [x] Code is added in right directory(Folder).
- [x] Brief Description of Solution is given.
- [x] Time and Space Complexity is mentioned
- [x] Code is well indented.
- [x] Comments are clearly written wherever necessary
- [x] Code is successfully compiled and passes all test cases.

### CODE REVIEW

File: 7. Dynamic Programming/cookie_collect/[CPP]cookie_collect_csubhradipta.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * @author [Subhradipta Choudhury](https://github.com/csubhradipta)
+ */
+
+#include <bits/stdc++.h> 
+using namespace std; 
+
+void solve(){
+    int n;
+    cin>>n;
+    int cookies[n+2][n], maxCookies[n];
```
Comment (anjali1903) [inline 7. Dynamic Programming/cookie_collect/[CPP]cookie_collect_csubhradipta.cpp]: Can you do this without using this extra 1D array?

Comment (cSubhradipta) [reply 7. Dynamic Programming/cookie_collect/[CPP]cookie_collect_csubhradipta.cpp]: > Can you do this without using this extra 1D array?

Yes Done!
Code Updated.

File: 7. Dynamic Programming/cookie_collect/[CPP]cookie_collect_csubhradipta.cpp [hunk 2]
```diff
@@ -17,26 +18,28 @@ using namespace std;
...
+        		cookies[i][j] += max(cookies[i - 1][j - 1], cookies[i][j - 1]);                         // for bottom row, left-down element is not there
 
-                if( j == n-1)
-                    maxCookies[i-1] = cookies[i][j];                                            // storing the elements of last column 
+        	else
+                cookies[i][j] += max({cookies[i - 1][j - 1], cookies[i][j - 1], cookies[i + 1][j - 1]});                // for intermediate rows, all element present
+                                                                                                            
+
+            if( j == n-1)
+                maxCookies[i] = cookies[i][j];                                  // storing the elements of last column
```
Comment (anjali1903) [inline 7. Dynamic Programming/cookie_collect/[CPP]cookie_collect_csubhradipta.cpp]: You're still using this array.

Comment (cSubhradipta) [reply 7. Dynamic Programming/cookie_collect/[CPP]cookie_collect_csubhradipta.cpp]: > You're still using this array.

I thought you're asking for the row of zeroes.
Code updated.

**Pull request merged**